services-camel-app/src/main/java/edu/unc/lib/boxc/services/camel/thumbnails/ThumbnailRouter.java

```diff
@@ -1,27 +1,46 @@
 package edu.unc.lib.boxc.services.camel.thumbnails;
 
+import edu.unc.lib.boxc.model.api.rdf.Cdr;
 import org.apache.camel.BeanInject;
+import org.apache.camel.PropertyInject;
 import org.apache.camel.builder.RouteBuilder;
 import org.slf4j.Logger;
 
+import static edu.unc.lib.boxc.services.camel.util.CdrFcrepoHeaders.CdrBinaryMimeType;
+import static edu.unc.lib.boxc.services.camel.util.CdrFcrepoHeaders.CdrBinaryPath;
 import static org.apache.camel.LoggingLevel.DEBUG;
 import static org.slf4j.LoggerFactory.getLogger;
 
 /**
- * Router for processing assigning thumbnails for works
+ * Router for processing thumbnails:
+ * Importing thumbnails for collections, folders, and admin units
+ * Assigning thumbnails for works
  *
  * @author snluong
  */
 public class ThumbnailRouter extends RouteBuilder {
     private static final Logger log = getLogger(ThumbnailRouter.class);
     @BeanInject(value = "thumbnailRequestProcessor")
     private ThumbnailRequestProcessor thumbnailRequestProcessor;
+    @BeanInject(value = "importThumbnailRequestProcessor")
+    private ImportThumbnailRequestProcessor importThumbnailRequestProcessor;
 
     @Override
     public void configure() throws Exception {
         from("{{cdr.thumbnails.stream.camel}}")
```

Comment:
After looking at the tests, I'm realizing that we wouldn't have the resourceType header available in practice. The queue here is receiving messages directly from ThumbnailRequestSender which doesn't set headers, plus headers are a camel thing so that activemq messages being sent can't actually set them. So I'm realizing that for the conditional in the router to work we'd have to have a processor that executed before the conditional in order to load the object being updated to get its resource type. Or we can use a second queue to separate the messages so that we wouldn't need to check the resource type.

I think there is also some disagreement between the `importThumbnail` controller and the logic here as well, since the controller doesn't actually enforce that the object being given a thumbnail is a folder/collection/adminunit. So if a work got a thumbnail imported, it'd go to the `otherwise` clause and throw an error during deserialization.  So we might either need to use two queues, or make sure the controller/services that send the messages always agree with the routing here.

Reply:
how do you specify two queues for the same sender? Like the default destination name is one queue and that is specified in the context file. do you need another Sender class for a different queue?